### ISSUE TITLE
How to make K-FAC work with BatchNorm layer?

### ISSUE THREAD

**Issue by knn1989** [ikostrikov/pytorch-a2c-ppo-acktr#85]
Hi Ilya,
Do you happen to know how (or where to start) to add BatchNorm support to your KFAC implementation?
Thanks!

**Comment by ikostrikov** [comment]
Hi,

KFAC itself performs some normalization similar to batch norm. It _whitens_ gradients instead of normalizing activations.

But if you want to implement BatchNorm anyway, here are some tips:
You need to care only about the parameters and not about normalization part of the BN. Now we just need to derive KFac approximation only for multiplication and addition. For addition use can just use the bias layer. For multiplication, it's essentially just a Linear layer with a diagonal matrix, so you can derive it from there.

**Comment by knn1989** [comment]
Thank you very much. This is very helpful.

By the way, can you let's know when you plan to add the asynchronous implementation to your kfac.

Thanks!
